File: LeetCode/1614.cpp
//Link: https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

class Solution {
public:
    int maxDepth(string s) {
        int depth = 0, max_depth = 0;
        for(int i = 0 ; i < s.length() ; i++) {
            if(s[i] == '(') depth += 1;
            else if(s[i] == ')') depth -= 1;
            max_depth = (depth > max_depth) ? depth : max_depth;
        }
        return max_depth;
    }
};

File: LeetCode/py/1614.py
#Link: https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

class Solution:
    def maxDepth(self, s: str) -> int:
        depth = max_depth = 0
        for ch in s:
            if ch == '(': depth += 1
            elif ch == ')': depth -= 1
            max_depth = max(depth,max_depth)
        return max_depth
        